agg-ast/schema_derivation/src/schema_derivation_tests/match_expr.rs

```diff
@@ -1675,3 +1675,1099 @@ mod misc_ops {
+        })),
+        input = r#"{"$match": {"$expr": {"$regexFindAll": {"input": "$foo", "regex": "$bar"}}}}"#,
+        starting_schema = Schema::Document(Document {
+            keys: map! {
+                "foo".to_string() => Schema::Any,
+                "bar".to_string() => Schema::Any,
+            },
+            required: set!("foo".to_string(), "bar".to_string()),
+            ..Default::default()
+        })
+    );
+    test_derive_schema_for_match_stage!(
+        regexfindall_eq_null,
+        expected = Ok(Schema::Document(Document {
+            keys: map! {
+                "foo".to_string() => Schema::Atomic(Atomic::String),
+                "bar".to_string() => Schema::AnyOf(set!(
+                    Schema::Atomic(Atomic::String),
+                    Schema::Atomic(Atomic::Regex),
+                )),
+            },
+            required: set!("foo".to_string(), "bar".to_string()),
+            ..Default::default()
+        })),
+        input = r#"{"$match": {"$expr": {"$eq": [null, {"$regexFindAll": {"input": "$foo", "regex": "$bar"}}]}}}"#,
+        starting_schema = Schema::Document(Document {
+            keys: map! {
+                "foo".to_string() => Schema::Any,
+                "bar".to_string() => Schema::Any,
+            },
+            required: set!("foo".to_string(), "bar".to_string()),
+            ..Default::default()
+        })
+    );
+    test_derive_schema_for_match_stage!(
+        regexfindall_opts,
+        expected = Ok(Schema::Document(Document {
+            keys: map! {
+                "foo".to_string() => Schema::Atomic(Atomic::String),
+                "bar".to_string() => Schema::AnyOf(set!(
+                    Schema::Atomic(Atomic::String),
+                    Schema::Atomic(Atomic::Regex),
+                )),
+                "car".to_string() => Schema::Atomic(Atomic::String),
+            },
+            required: set!("foo".to_string(), "bar".to_string(), "car".to_string()),
+            ..Default::default()
+        })),
+        input = r#"{"$match": {"$expr": {"$regexFindAll": {"input": "$foo", "regex": "$bar", "options": "$car"}}}}"#,
+        starting_schema = Schema::Document(Document {
+            keys: map! {
+                "foo".to_string() => Schema::Any,
+                "bar".to_string() => Schema::Any,
+                "car".to_string() => Schema::Any,
+            },
+            required: set!("foo".to_string(), "bar".to_string()),
+            ..Default::default()
+        })
+    );
+    test_derive_schema_for_match_stage!(
+        regexfindall_opts_eq_null,
+        expected = Ok(Schema::Document(Document {
+            keys: map! {
+                "foo".to_string() => Schema::Atomic(Atomic::String),
+                "bar".to_string() => Schema::AnyOf(set!(
+                    Schema::Atomic(Atomic::String),
+                    Schema::Atomic(Atomic::Regex),
+                )),
+                "car".to_string() => Schema::Atomic(Atomic::String),
```

Comment:
nit: I think car can actually be nullish while still being valid -- ie its `STRING_OR_NULLISH` in all nullability contexts.